psi-notify: init at 1.2.1

eduarrrd opened the pull request:
###### Motivation for this change

New package I use.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

eduarrrd committed:
psi-notify: init at 1.2.1

figsoda commented on pkgs/applications/misc/psi-notify/default.nix:
```diff
@@ -0,0 +1,36 @@
+    sha256 = "0hn37plim1smmlrjjmz8kybyms8pz3wxcgf8vmqjrsqi6bfcym7g";
+  };
+
+  buildInputs = [ systemd libnotify pkg-config ];
```
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ libnotify systemd ];
```

figsoda commented on pkgs/applications/misc/psi-notify/default.nix:
```diff
@@ -0,0 +1,36 @@
+  };
+
+  buildInputs = [ systemd libnotify pkg-config ];
+  buildPhase = ''
+    make
+  '';
```
this is the default
```suggestion

```

figsoda commented on pkgs/applications/misc/psi-notify/default.nix:
```diff
@@ -0,0 +1,36 @@
+  buildPhase = ''
+    make
+  '';
+  installPhase = ''
+    install -D psi-notify $out/bin/psi-notify
+    substituteInPlace psi-notify.service --replace psi-notify $out/bin/psi-notify
+    install -D psi-notify.service $out/share/systemd/user/psi-notify.service
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall
    install -D psi-notify $out/bin/psi-notify
    substituteInPlace psi-notify.service --replace psi-notify $out/bin/psi-notify
    install -D psi-notify.service $out/share/systemd/user/psi-notify.service
    runHook postInstall
  '';
```

figsoda commented on pkgs/applications/misc/psi-notify/default.nix:
```diff
@@ -0,0 +1,36 @@
+    license = licenses.mit;
+    homepage = "https://github.com/cdown/psi-notify";
+    platforms = platforms.linux;
+    maintainers = [ maintainers.eduarrrd ];
```
```suggestion
    maintainers = with maintainers; [ eduarrrd ];
```

figsoda commented:
upstream has a togglable feature `WANT_SD_NOTIFY` can be set to 0 and make the libsystemd dependency optional
https://github.com/cdown/psi-notify/blob/1.2.1/Makefile#L4

eduarrrd commented:
Addressed the suggestions, they make sense to me.

> upstream has a togglable feature `WANT_SD_NOTIFY` that can be set to 0 and make the libsystemd dependency optional
> https://github.com/cdown/psi-notify/blob/1.2.1/Makefile#L4
> 
> maybe it could be added to this package too?

Possibly, but I don't intend to use it without that. We can discuss this once there's a user who intends to use it this way.

figsoda commented:
fair enough

figsoda commented:
Result of `nixpkgs-review pr 139335` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>psi-notify</li>
  </ul>
</details>
